addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/EvohomeApiClientV2.java

```diff
@@ -0,0 +1,252 @@
+        url = String.format(url, tcsId);
+        apiAccess.doAuthenticatedRequest(HttpMethod.PUT, url, null, new Mode(mode), null);
+    }
+
+    @Override
+    public void setHeatingZoneOverride(String zoneId, double setPoint) {
+        setHeatingZoneOverride(zoneId, new HeatSetPoint(setPoint));
+    }
+
+    @Override
+    public void cancelHeatingZoneOverride(String zoneId) {
+        setHeatingZoneOverride(zoneId, new HeatSetPoint());
+    }
+
+    private void setHeatingZoneOverride(String zoneId, HeatSetPoint heatSetPoint) {
+        String url = EvohomeApiConstants.URL_V2_BASE + EvohomeApiConstants.URL_V2_HEAT_SETPOINT;
+        url = String.format(url, zoneId);
+        apiAccess.doAuthenticatedRequest(HttpMethod.PUT, url, null, heatSetPoint, null);
+    }
+
+    private UserAccount requestUserAccount() {
+        String url = EvohomeApiConstants.URL_V2_BASE + EvohomeApiConstants.URL_V2_ACCOUNT;
+
+        UserAccount userAccount = new UserAccount();
+        userAccount = apiAccess.doAuthenticatedRequest(HttpMethod.GET, url, null, null, userAccount);
+
+        return userAccount;
+    }
+
+    private Locations requestLocations() {
+        Locations locations = null;
+        if (useraccount != null) {
+            String url = EvohomeApiConstants.URL_V2_BASE + EvohomeApiConstants.URL_V2_INSTALLATION_INFO;
+            url = String.format(url, useraccount.userId);
+
+            locations = new Locations();
+            locations = apiAccess.doAuthenticatedRequest(HttpMethod.GET, url, null, null, locations);
+        }
+
+        return locations;
+    }
+
+    private LocationsStatus requestLocationsStatus() {
+        LocationsStatus locationsStatus = null;
+
+        if (locations != null) {
+            locationsStatus = new LocationsStatus();
+
+            for (Location location : locations) {
+                String url = EvohomeApiConstants.URL_V2_BASE + EvohomeApiConstants.URL_V2_LOCATION_STATUS;
+                url = String.format(url, location.locationInfo.locationId);
+                LocationStatus status = new LocationStatus();
+                status = apiAccess.doAuthenticatedRequest(HttpMethod.GET, url, null, null, status);
+                locationsStatus.add(status);
+            }
+        }
+
+        return locationsStatus;
+    }
+
+    private boolean authenticate(String credentials, String grantType) {
+        Authentication authentication = new Authentication();
+
+        String data = credentials + "&" + "Host=rs.alarmnet.com%2F&" + "Pragma=no-cache&"
+                + "Cache-Control=no-store+no-cache&"
+                + "scope=EMEA-V1-Basic+EMEA-V1-Anonymous+EMEA-V1-Get-Current-User-Account&" + "grant_type=" + grantType
+                + "&" + "Content-Type=application%2Fx-www-form-urlencoded%3B+charset%3Dutf-8&"
+                + "Connection=Keep-Alive";
+
+        HashMap<String, String> headers = new HashMap<String, String>();
```

Comment:
Use the interface instead of the concrete class and use the diamond operator, ie:

```java
Map<String, String> headers = new HashMap<>();
```